scripts/morty/analysis.py

```diff
@@ -0,0 +1,273 @@
+import solvers
+import numpy as np
+from data import DataHandler
+from scipy import integrate
+
+
+class AnalysisCollection:
+    def __init__(self, analysis_params, run_params, data_params):
+        """
+        This class contains a collection of analysis procedures which
+        can be run.
+        The data is returned in an easily-plottable dictionary as
+        well as saved to
+        text files for future analysis.
+
+        Parameters
+        ----------
+        analysis_params : dict
+            key : str
+                Name of analysis parameter
+        run_params : dict
+            key : str
+                Name of run parameter
+        data_params : dict
+            ket : str
+                Name of data parameter
+        """
+        self.analysis_params = analysis_params
+        self.run_params = run_params
+        self.data_params = data_params
+        return
+
+    def _time_lab(self):
+        """
+        Determines time scaling to use and labelling
+
+        Returns
+        -------
+        scaling_factor : float
+            Factor to scale time in (s) by
+        xlab : str
+            Time label to use
+        """
+        tf = self.run_params['final_time']
+        second = 1
+        millisecond = 1e-3 * second
+        minute = 60 * second
+        hour = 60 * minute
+        day = 24 * hour
+        year = 365.25 * day
+        time_list = [year, day, hour, minute, second, millisecond]
+        label_list = ['yr', 'd', 'hr', 'min', 's', 'ms']
+        for ti, time in enumerate(time_list):
+            if time > tf:
+                continue
+            xlab = f'Time [{label_list[ti]}]'
+            scaling_factor = 1 / time
+            return scaling_factor, xlab
+
+    def _method_change(
+            self,
+            methods,
+            method_name,
+            xfactor=1,
+            methodname_extension=''):
+        """
+        Used to handle concentration plotting comparisons with a single
+            method change.
+
+        Parameters
+        ----------
+        methods : list
+            List of methods to apply
+        method_name : str
+            Key for dictionary to apply method to `run_params`
+        xfactor : float
+            Value to scale `x_vals` by
+        methodname_extension : str (optional)
+            String to add to method name in plotting
+
+        Returns
+        -------
+        xs : list of list
+            Contains the `x` values to be plotted
+        ys : list of list
+            Contains the `y` values to be plotted
+        labs : list of list
+            Contains the ordered labels to use in plotting
+
+
+        """
+        xs, ys, labs = [], [], []
```

Comment:
I think labels is a better variable name here. It's only two extra letters and quite a lot clearer what you're doing.